### PR TITLE
docs: Improve instructions for using update_deps

### PR DESCRIPTION
**Pull request by karl-lunarg**
[skip-ci]

### CODE REVIEW

File: BUILD.md [hunk 1]
```diff
@@ -134,22 +134,41 @@ directories and place them in any location.
...
 
-Here is a usage example for this repository:
+For all platforms, start with:
 
     git clone git@github.com:KhronosGroup/Vulkan-ValidationLayers.git
     cd Vulkan-ValidationLayers
     mkdir build
     cd build
+
+For 64-bit Linux and MacOS, continue with:
```
Comment (karl-lunarg) [inline BUILD.md]: @jeremyk-lunarg Can you please double check this on MacOS?

**Comment by karl-lunarg** [comment]
Should have put `[ci skip]` in the commit message to skip CI for this doc-only change.  Oh well.

File: BUILD.md [hunk 2]
```diff
@@ -134,22 +134,41 @@ directories and place them in any location.
...
 ### Building Dependent Repositories with Known-Good Revisions
 
-There is a Python utility script, `scripts/update_deps.py`, that you can use
-to gather and build the dependent repositories mentioned above. This program
-also uses information stored in the `scripts/known-good.json` file to checkout
-dependent repository revisions that are known to be compatible with the
-revision of this repository that you currently have checked out.
+There is a Python utility script, `scripts/update_deps.py`, that you can use to
+gather and build the dependent repositories mentioned above. This script uses
+information stored in the `scripts/known-good.json` file to checkout dependent
```
Comment (lunarpapillo) [inline BUILD.md]: `known-good.json` => `known_good.json`
checkout => check out

File: BUILD.md [hunk 3]
```diff
@@ -134,22 +134,41 @@ directories and place them in any location.
...
+repository revisions that are known to be compatible with the revision of this
+repository that you currently have checked out. As such, this script is useful
+as a quick start tool for common use cases and default configurations.
```
Comment (lunarpapillo) [inline BUILD.md]: quick start => quick-start

File: BUILD.md [hunk 4]
```diff
@@ -134,22 +134,41 @@ directories and place them in any location.
...
     mkdir build
     cd build
+
+For 64-bit Linux and MacOS, continue with:
+
     ../scripts/update_deps.py
     cmake -C helper.cmake ..
     cmake --build .
 
+For 64-bit Windows, continue with:
```
Comment (lunarpapillo) [inline BUILD.md]: The Windows build assumes only one valid build system, right?  Or you'd have to pass the `--generator` switch to `update_deps.py`, yes?

I'm uncertain as to whether it's worthwhile to point this out; I don't think it happens often, but it did happen to me...

File: BUILD.md (hunk 2 above)
Comment (karl-lunarg) [reply BUILD.md]: fixed and fixed

File: BUILD.md (hunk 3 above)
Comment (karl-lunarg) [reply BUILD.md]: fixed

File: BUILD.md (hunk 4 above)
Comment (karl-lunarg) [reply BUILD.md]: The point of this section of the doc is to give a quick and short set of instructions that work in most cases.  If we start explaining all possible permutations and situations, this goal gets lost.  There are repeated notes here and near here that point the reader to other parts of the instructions should they need to handle something differently.

File: BUILD.md (hunk 1 above)
Comment (jeremyk-lunarg) [reply BUILD.md]: These instructions work on MacOS.

**Comment by jjulianoatnv** [comment]
This is a good improvement. It would have saved me some wasted effort when I built Vulkan-Tools recently on a new computer. I had made a mistake of specifying x64 when runninn update_deps.py, but then forgetting to specify x64 when doing the initial cmake command. Spelling out the steps as you are doing in this PR decreases probability that someone else will make the same mistake.

**Pull request merged**